### ISSUE TITLE
`path_is_subpath()` is broken on Windows

### ISSUE THREAD

**Issue by mih** [datalad/datalad#2564]
```
[00:15:30] ======================================================================
[00:15:30] FAIL: datalad.tests.test_utils.test_path_is_subpath
[00:15:30] ----------------------------------------------------------------------
[00:15:30] Traceback (most recent call last):
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\nose\case.py", line 198, in runTest
[00:15:30]     self.test(*self.arg)
[00:15:30]   File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\test_utils.py", line 775, in test_path_is_subpath
[00:15:30]     ok_(path_is_subpath('/a/b', '/a'))
[00:15:30] AssertionError
```

**Comment by yarikoptic** [comment]
if we are to stay with native paths on the OS, then it is not a totally appropriate test for Windows since `/` is then defined (at best) up to the drive only and without a drive letter you cannot really get an answer

**Comment by mih** [comment]
https://github.com/datalad/datalad-revolution uses pathlib to avoid this code entirely

**Comment by mih** [comment]
The release of 0.12 with this fix is imminent (#3993) closing.

**Comment by mih** [comment]
Well, the issue with the test itself remains, actually. It is still featuring `@skip_if_on_windows`. Although the usage of helpers like this is much reduced, we simply ignore whether they work or not work on windows.

This is quite bad, because, for example `path_is_subpath()` is used in central code, like the result processing.

IMHO It should really be replaced. All of what `path_is_subpath()` is doing is

```py
Path(prefix) in Path(path).parents
```
